stdlib/public/Backtracing/Dwarf.swift

```diff
@@ -1553,13 +1553,32 @@ struct DwarfReader<S: DwarfSource> {
       return
     }
 
-    let refAttrs = try readDieAttributes(
+    var refAttrs = try readDieAttributes(
       at: &cursor,
       unit: unit,
       abbrevInfo: abbrevInfo,
       shouldFetchIndirect: true
     )
 
+    if let specificationVal = refAttrs[.DW_AT_specification],
+       case let .reference(specification) = specificationVal {
+      cursor = ImageSourceCursor(source: infoSection,
+                                 offset: specification)
+      abbrev = try cursor.readULEB128()
+      if abbrev == 0 {
+        return
+      }
+      guard let abbrevInfo = unit.abbrevs[abbrev] else {
+        throw DwarfError.missingAbbrev(abbrev)
+      }
+      refAttrs = try readDieAttributes(
```

Comment:
Should we check
```
    let tag = abbrevInfo.tag
    if tag != .DW_TAG_subprogram {
       return
    }
```
here as well?

Reply:
Thanks for the review, but would you prefer to throw an exception?

Reply:
`throw` in Swift doesn't throw exceptions per se — that's Swift error handling, and it doesn't really work the same way. 

As for whether it should `return` or `throw`, if you `throw` in that function, then there will be no inline frame data available for that DWARF unit (at all), even if the rest is just fine. The couple of `throw`s that exist in that function are for cases where there's something awful wrong (like a missing abbrev, or a crazy address size), rather than cases where we've just failed to properly understand some DWARF data. So I think `return` is the right thing here.

Reply:
Sorry, I might've miscommunicated. I mean error handling using `throw`.
Yes, using return would be better. It's good to provide as much information as possible when something goes wrong. Thanks.